src/datumaro/plugins/data_formats/image_dir.py

```diff
@@ -27,30 +33,57 @@ def build_cmdline_parser(cls, **kwargs):
         )
         return parser
 
+    @classmethod
+    def detect(cls, context: FormatDetectionContext) -> Optional[FormatDetectionConfidence]:
+        path = Path(context.root_path)
+        for item in path.iterdir():
+            if item.is_dir():
+                if item.name not in SPECIAL_MACOS_FOLDERS:
+                    context.fail("Only flat image directories are supported")
+            elif item.suffix.lower() not in IMAGE_EXTENSIONS:
+                context.fail(f"File {item} is not an image.")
```

Comment:
It looks like `contains_only_images` should also be used here, just without recursion inside.